src/combinatorics/simple_graph/strongly_regular.lean

```diff
@@ -41,27 +40,133 @@ A graph is strongly regular with parameters `n k l m` if
  * every pair of adjacent vertices has `l` common neighbors
  * every pair of nonadjacent vertices has `m` common neighbors
 -/
-structure is_SRG_of (n k l m : ℕ) : Prop :=
+structure is_SRG_with (n k l m : ℕ) : Prop :=
 (card : fintype.card V = n)
 (regular : G.is_regular_of_degree k)
 (adj_common : ∀ (v w : V), G.adj v w → fintype.card (G.common_neighbors v w) = l)
-(nadj_common : ∀ (v w : V), ¬ G.adj v w ∧ v ≠ w → fintype.card (G.common_neighbors v w) = m)
+(nadj_common : ∀ (v w : V), v ≠ w ∧ ¬G.adj v w → fintype.card (G.common_neighbors v w) = m)
 
-open finset
+variables {n k l m : ℕ}
+
+/-- Empty graphs are strongly regular. Note that `l` can take any value
+  for empty graphs, since there are no pairs of adjacent vertices. -/
+lemma bot_strongly_regular :
+  (⊥ : simple_graph V).is_SRG_with (fintype.card V) 0 l 0 :=
+{ card := rfl,
+  regular := bot_degree,
+  adj_common := λ v w h, h.elim,
+  nadj_common := λ v w h, begin
+    simp only [finset.card_eq_zero, finset.filter_congr_decidable, fintype.card_of_finset],
+    ext,
+    simp [mem_common_neighbors],
+  end }
 
-/-- Complete graphs are strongly regular. Note that the parameter `m` can take any value
-  for complete graphs, since there are no distinct pairs of nonadjacent vertices. -/
-lemma complete_strongly_regular (m : ℕ) :
-  (⊤ : simple_graph V).is_SRG_of (fintype.card V) (fintype.card V - 1) (fintype.card V - 2) m :=
+/-- Complete graphs are strongly regular. Note that `m` can take any value
+  for complete graphs, since there are no distinct pairs of non-adjacent vertices. -/
+lemma top_strongly_regular :
+  (⊤ : simple_graph V).is_SRG_with (fintype.card V) (fintype.card V - 1) (fintype.card V - 2) m :=
 { card := rfl,
-  regular := complete_graph_degree,
-  adj_common := λ v w (h : v ≠ w),
-    begin
-      simp only [fintype.card_of_finset, mem_common_neighbors, filter_not, ←not_or_distrib,
-                 filter_eq, filter_or, card_univ_diff, mem_univ, if_pos, ←insert_eq, top_adj],
-      rw [card_insert_of_not_mem, card_singleton],
-      simp [h]
-    end,
-  nadj_common := λ v w (h : ¬(v ≠ w) ∧ _), (h.1 h.2).elim }
+  regular := complete_graph_is_regular,
+  adj_common := λ v w h, begin
+    rw card_common_neighbors_top,
+    exact h,
+  end,
+  nadj_common := λ v w h, false.elim $ by simpa using h }
+
+lemma card_neighbor_finset_union_eq {v w : V} (h : G.is_SRG_with n k l m) :
```

Comment:
For dot notation:
```suggestion
lemma is_SRG_with.card_neighbor_finset_union_eq {v w : V} (h : G.is_SRG_with n k l m) :
```
(Make sure to switch to dot notation where it's used, if possible.)